prow/plugins/caicloud-cla/cla.go

```diff
@@ -0,0 +1,177 @@
+}
+
+// 1. Check that the PR event received from the webhook.
+// 2. Use the github search API to get the PR which match the commit hash corresponding to the PR event.
+// 3. For PR that matches, check that the PR's HEAD commit hash against the commit hash.
+// 4. Set the corresponding CLA label if needed.
+func handlePR(gc githubClient, log *logrus.Entry, pe github.PullRequestEvent) error {
+	if !isPRChanged(pe) {
+		return nil
+	}
+
+	var (
+		org  = pe.PullRequest.Base.Repo.Owner.Login
+		repo = pe.PullRequest.Base.Repo.Name
+		num  = pe.PullRequest.Number
+	)
+
+	log.Info("PR labels may be out of date. Getting pull request info.")
+	pr, err := gc.GetPullRequest(org, repo, num)
+	if err != nil {
+		log.WithError(err).Warningf("Unable to fetch PR-%d from %s/%s.", num, org, repo)
+		return nil
+	}
+
+	number := pr.Number
+	isMember, err := gc.IsMember(org, pr.User.Login)
+	if err != nil {
+		log.WithError(err).Errorf("Error from IsMember(%q of org %q).", pr.User.Login, org)
+	}
+	if !isMember {
+		if err := gc.AddLabel(org, repo, number, claYesLabel); err != nil {
+			log.WithError(err).Warningf("Could not add %s label.", claYesLabel)
+		}
+		return nil
+	}
+
+	commit, err := gc.GetCommit(org, repo, pr.Head.SHA)
+	if err != nil {
+		log.WithError(err).Warningf("Unable to get commit-%s from %s/%s.", pr.Head.SHA, org, repo)
+		return nil
+	}
+
+	claReady := false
+	if emailRe.MatchString(commit.Author.Email) && commit.Author.Email == commit.Committer.Email {
+		claReady = true
+	}
+
+	labels, err := gc.GetIssueLabels(org, repo, num)
+	if err != nil {
+		log.Warnf("while retrieving labels, error: %v", err)
+	}
+
+	hasCaicloudYes := false
+	hasCaicloudNo := false
+	for _, label := range labels {
+		if label.Name == claYesLabel {
+			hasCaicloudYes = true
+			continue
+		}
+		if label.Name == claNoLabel {
+			hasCaicloudNo = true
+			continue
+		}
+
+	}
+
+	if hasCaicloudYes && claReady {
+		// Nothing to update.
+		log.Infof("PR has up-to-date %s label.", claYesLabel)
+		return nil
+	}
+
+	if hasCaicloudNo && !claReady {
+		// Nothing to update.
+		log.Infof("PR has up-to-date %s label.", claNoLabel)
+		return nil
+	}
+
+	if hasCaicloudNo && claReady {
+		if err := gc.RemoveLabel(org, repo, number, claNoLabel); err != nil {
+			log.WithError(err).Warningf("Could not remove %s label.", claNoLabel)
+		}
+		if err := gc.AddLabel(org, repo, number, claYesLabel); err != nil {
+			log.WithError(err).Warningf("Could not add %s label.", claYesLabel)
+		}
+		return nil
+	}
+
+	if hasCaicloudYes {
+		if err := gc.RemoveLabel(org, repo, number, claYesLabel); err != nil {
+			log.WithError(err).Warningf("Could not remove %s label.", claYesLabel)
+		}
+		if err := gc.AddLabel(org, repo, number, claNoLabel); err != nil {
+			log.WithError(err).Warningf("Could not add %s label.", claNoLabel)
+		}
+	}
+	if err := gc.CreateComment(org, repo, number, fmt.Sprintf(caicloudclaNotFoundMessage, plugins.AboutThisBot)); err != nil {
```

Comment:
initially, a PR has no `hasCaicloudYes` and `hasCaicloudNo` (both will be false), based on your logic here,  even if i'm a caicloud member, i still get this cla not found message?